code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -123,6 +133,133 @@
+
+	playsound(xeno_owner.loc, 'sound/voice/alien/queen_heal_screech.ogg', 75, 0)
+	xeno_owner.visible_message(span_xenohighdanger("\The [xeno_owner] emits an ear-splitting guttural roar!"))
+
+	succeed_activate()
+	add_cooldown()
+
+	var/datum/action/ability/activable/xeno/normal_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/screech]
+	if(normal_screech)
+		normal_screech.add_cooldown(30 SECONDS)
+	var/datum/action/ability/activable/xeno/support_screech/plasma_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/support_screech/plasma]
+	if(plasma_screech)
+		plasma_screech.add_cooldown(30 SECONDS)
+	var/datum/action/ability/activable/xeno/support_screech/frenzy_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/support_screech/frenzy]
+	if(frenzy_screech)
+		frenzy_screech.add_cooldown(30 SECONDS)
+
+/datum/action/ability/activable/xeno/support_screech/plasma
+	name = "Plasma Screech"
+	action_icon_state = "plasma_screech"
+	action_icon = 'icons/Xeno/actions/queen.dmi'
+	desc = "Screech that increases plasma regeneration for nearby xenos."
+	var/bonus_regen = 0.5
+	var/duration = 30 SECONDS
+	keybind_flags = ABILITY_KEYBIND_USE_ABILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PLASMA_SCREECH,
+	)
+
+/datum/action/ability/activable/xeno/support_screech/plasma/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/queen/xeno_owner = owner
+
+	for(var/mob/living/carbon/xenomorph/affected_xeno in cheap_get_xenos_near(xeno_owner, screech_range))
+		if(!(affected_xeno.xeno_caste.can_flags & CASTE_CAN_BE_GIVEN_PLASMA))
+			continue
+		affected_xeno.apply_status_effect(/datum/status_effect/plasma_surge, affected_xeno.xeno_caste.plasma_max / 2, bonus_regen, duration)
+
+	playsound(xeno_owner.loc, 'sound/voice/alien/queen_plasma_screech.ogg', 75, 0)
+	xeno_owner.visible_message(span_xenohighdanger("\The [xeno_owner] emits an ear-splitting guttural roar!"))
+
+	succeed_activate()
+	add_cooldown()
+
+	var/datum/action/ability/activable/xeno/normal_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/screech]
+	if(normal_screech)
+		normal_screech.add_cooldown(30 SECONDS)
+	var/datum/action/ability/activable/xeno/support_screech/heal_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/support_screech/heal]
+	if(heal_screech)
+		heal_screech.add_cooldown(30 SECONDS)
+	var/datum/action/ability/activable/xeno/support_screech/frenzy_screech = xeno_owner.actions_by_path[/datum/action/ability/activable/xeno/support_screech/frenzy]
+	if(frenzy_screech)
+		frenzy_screech.add_cooldown(30 SECONDS)
+
+/datum/action/ability/activable/xeno/support_screech/frenzy
+	name = "Frenzy Screech"
+	action_icon_state = "frenzy_screech"
+	action_icon = 'icons/Xeno/actions/queen.dmi'
+	desc = "Screech that increases damage for nearby xenos."
+	var/buff_duration = 30 SECONDS
+	var/buff_damage_modifier = 0.1
```

Comment:
Move it after the `keybinding_signals` list and document it. This could also be a local define